ChangeLog.md

```diff
@@ -22,6 +22,8 @@ See docs/process.md for more on how version tagging works.
 ------
 - Fix crash, introduced in 3.1.11, which occurred when using pointer types
   (types ending in `*`) with getValue/setValue library functions. (#17028)
+- Fix possible deadlock in multi-threaded builds that use EXIT_RUNTIME=0
+  which was introduced in 3.1.3 as part of #16130.
```

Comment:
IIUC those deadlocks only happened when `ASSERTIONS`? (as that's when `checkUnflushedContent` is included) Might be worth mentioning here to give more context.